TPP.Persistence.MongoDB/Repos/BadgeRepo.cs

```diff
@@ -103,8 +103,10 @@ public async Task<List<Badge>> FindAllByCustom(string? userId = null, PkmnSpecie
                 filter &= Builders<Badge>.Filter.Eq(b => b.Form, form);
             if (source != null)
                 filter &= Builders<Badge>.Filter.Eq(b => b.Source, source);
-            if (shiny != null)
-                filter &= Builders<Badge>.Filter.Eq(b => b.Shiny, shiny);
+            if (shiny == true)
+                filter &= Builders<Badge>.Filter.Eq(b => b.Shiny, true);
+            else
+                filter &= Builders<Badge>.Filter.Ne(b => b.Shiny, true);
```

Comment:
Pls add a short comment along the lines of `cannot test for == false, because that would not match null`